doors_dashboards/components/scatterplot.py

```diff
@@ -807,167 +910,224 @@ def stores_to_labels_maingroup(general_data):
                 return tuple(results)
 
         @callback(
-            [Output(group_dropdown_menu, 'label')
-             for group_dropdown_menu in group_dropdown_menus],
-            Input(f"{dashboard_id}-{GENERAL_STORE_ID}", 'data'),
-            prevent_initial_call=True
+            [
+                Output(group_dropdown_menu, "label")
+                for group_dropdown_menu in group_dropdown_menus
+            ],
+            Input(f"{dashboard_id}-{GENERAL_STORE_ID}", "data"),
+            prevent_initial_call=True,
         )
         def stores_to_labels_group(general_data):
             general_data = general_data or {}
             results = []
-            for group_dropdown_menu_id, group_dropdown_menu \
-                    in self.group_dropdown_menus.items():
-                c, main_group = \
-                    self.decode_group_dropdown(group_dropdown_menu_id)
-                default_group = \
-                    self._get_group_values_for_main_group(c, main_group)[0]
-                group = general_data.get(GROUPS_SECTION, {}).get(c, {}). \
-                    get(GROUP, default_group)
+            for (
+                group_dropdown_menu_id,
+                group_dropdown_menu,
+            ) in self.group_dropdown_menus.items():
+                c, main_group = self.decode_group_dropdown(group_dropdown_menu_id)
+                default_group = self._get_group_values_for_main_group(c, main_group)[0]
+                group = (
+                    general_data.get(GROUPS_SECTION, {})
+                    .get(c, {})
+                    .get(GROUP, default_group)
+                )
```

Comment:
![image](https://github.com/bcdev/doors-dashboards/assets/144326622/038af289-7ed7-4810-9567-eb024177b42c)

Similarly for group  also label is not updating sometimes